plugins/src/main/kotlin/io/github/yonggoose/organizationdefaults/ArtifactCheckPlugin.kt

```diff
@@ -2,93 +2,326 @@ package io.github.yonggoose.organizationdefaults
+        }
 
-                // License validation
-                pom.licenses.forEach { license ->
-                    if (license.licenseType.isNullOrBlank()) {
-                        errors.add("Invalid license: License name is required.")
-                    }
-                }
+        if (pom.artifactId.isNullOrBlank()) {
+            errors.add("Invalid artifactId: Pom must not be null or blank.")
+        }
 
-                // Developer Info validation
-                pom.developers.forEach { developer ->
-                    if (developer.name.isNullOrBlank()) {
-                        errors.add("Invalid developer: Developer name is required.")
-                    }
-                    if (developer.email.isNullOrBlank()) {
-                        errors.add("Invalid developer: Developer email is required.")
-                    }
-                    if (developer.organization.isNullOrBlank()) {
-                        errors.add("Invalid developer: Organization is required.")
-                    }
-                    if (developer.organizationUrl.isNullOrBlank()) {
-                        errors.add("Invalid developer: Organization URL is required.")
-                    }
-                }
+        val version = pom.version
+        if (version == null || (version.isBlank() || version.endsWith("-SNAPSHOT"))) {
+            errors.add("Invalid version: The version must not be null, blank, or end with '-SNAPSHOT'.")
+        }
 
-                // SCM Information validation
-                pom.scm?.let { scm ->
-                    if (scm.connection.isNullOrBlank()) {
-                        errors.add("Invalid SCM: Read-only connection is required.")
-                    }
-                    if (scm.developerConnection.isNullOrBlank()) {
-                        errors.add("Invalid SCM: Read/write connection is required.")
-                    }
-                    if (scm.url.isNullOrBlank()) {
-                        errors.add("Invalid SCM: Web interface URL is required.")
-                    }
-                }
+        // Project Information validation
+        if (pom.name.isNullOrBlank()) {
+            errors.add("Invalid name: Project name is required.")
+        }
+        if (pom.description.isNullOrBlank()) {
+            errors.add("Invalid description: Project description is required.")
+        }
+        if (pom.url.isNullOrBlank()) {
+            errors.add("Invalid url: Project URL is required.")
+        }
 
-                // Throw exception if there are errors
-                if (errors.isNotEmpty()) {
-                    throw IllegalArgumentException("Validation failed:\n${errors.joinToString("\n")}")
-                }
+        // License validation
+        pom.licenses.forEach { license ->
+            if (license.licenseType.isNullOrBlank()) {
+                errors.add("Invalid license: License name is required.")
+            }
```

Comment:
What is `licenseType` ? It doesn't seem to be part of [the spec](https://maven.apache.org/pom.html#Licenses)?